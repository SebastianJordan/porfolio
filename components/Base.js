import styles from "../styles/Base.module.css";
export function Base(props) {
  return <main className={styles.page}>{props.children}</main>;
}
