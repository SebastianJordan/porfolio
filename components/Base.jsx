import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/Base.module.css';
export function Base(props) {
  return <main className={styles.page}>{props.children}</main>;
}

Base.propTypes = {
  children: PropTypes.node.isRequired,
};
