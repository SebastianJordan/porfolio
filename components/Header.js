export function Header() {
  return (
    <>
      <meta charset="UTF-8" />
      <link rel="icon" type="image/svg+xml" href="/src/images/icon.ico" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <meta
        name="description"
        content="Hello, I am a system engineer with more than 5 years working on technology projects in the backend area (Java) such as frontend (Angular and React) and development of hybrid and native mobile applications (Ionic and ReactNative) applying an agile scrum methodology"
      />
      <meta name="author" content="Sebastian Jordan Frontend" />
      <meta
        property="og:title"
        content="Hello to personal website where I show my skills and resume"
      />
      <meta
        property="og:description"
        content="My personal portfolio where I show my technical skills at the fullstack level"
      />
      <meta property="og:image" content="src/images/icon.png" />

      <meta name="twitter:card" content="summary_large_image" />
      <meta name="twitter:site" content="@sjmPanama" />
      <meta
        name="twitter:title"
        content="Hello to personal website where I show my skills and resume"
      />
      <meta
        name="twitter:description"
        content="My personal portfolio where I show my technical skills at the fullstack level"
      />
      <meta name="twitter:image" content="src/images/icon.png" />
      <title>
        Hi, i&apos;m Sebasti&aacute;n Jord&aacute;n this is my portfolio
      </title>
      {/* Global Site Tag (gtag.js) - Google Analytics */}
      <script
        async
        src={`https://www.googletagmanager.com/gtag/js?id=${process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS}`}
      />
      <script
        dangerouslySetInnerHTML={{
          __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', '${process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS}', {
              page_path: window.location.pathname,
            });
          `,
        }}
      />
    </>
  );
}
