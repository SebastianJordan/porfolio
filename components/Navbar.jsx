import React from 'react';
import styles from '../styles/Navbar.module.css';
import PropTypes from 'prop-types';
export function Navbar({ showMenu }) {
  return (
    <nav className={styles.box}>
      <h2 className={styles.logo}>S</h2>
      {showMenu && (
        <>
          <ul className={styles.menu}>
            <li className={styles.item}>About</li>
            <li className={styles.item}>Experience</li>
            <li className={styles.item}>Project</li>
          </ul>
          <button className={styles.btn}>Contact</button>
        </>
      )}
    </nav>
  );
}

Navbar.propTypes = {
  showMenu: PropTypes.bool,
};
