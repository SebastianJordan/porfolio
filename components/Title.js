import styles from "../styles/Title.module.css";

export default function Title(props) {
  return (
    <>
      <div className={styles.groupPoint}>
      </div>
      <h1 className={styles.title}>
      { props.title.map((i)=>
         <>{i}<br/></>
      )}
      </h1>
      <div className={styles.description}>
        {props.children}
      </div>
    </>
  );
}
