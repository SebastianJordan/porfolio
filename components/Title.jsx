import React from 'react';
import styles from '../styles/Title.module.css';
import PropTypes from 'prop-types';

export default function Title(props) {
  const { title, children } = props;
  return (
    <>
      <div className={styles.groupPoint}></div>
      <h1 className={styles.title}>
        {title.map((i) => (
          <>
            {i}
            <br />
          </>
        ))}
      </h1>
      <div className={styles.description}>{children}</div>
    </>
  );
}

Title.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired,
};
