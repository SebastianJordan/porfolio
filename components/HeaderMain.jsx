import Title from "./Title";
import styles from "../styles/UnderConstruction.module.css";
export function HeaderMain() {
  return (
    <section className={styles.slideInfo}>
      <Title
        title={["Hi", "Iâ€™m Sebastian", "Front-end developer"]}
        showBg={true}
      ></Title>
      <a href="#" className="btn-basic mr-32">
        View resume
      </a>
      <a href="#" className="btn-link">
        Github Projects{" "}
      </a>
    </section>
  );
}
