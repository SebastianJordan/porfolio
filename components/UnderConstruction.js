import styles from "../styles/UnderConstruction.module.css";
import { useEmail } from "../hooks/useEmail";
import Title from "./Title";

export function UnderConstruction() {
  const [btn, emailInput, loading, onChangeForm, onSendEmail] = useEmail({
    btn: "Notify me",
    emailInput: "",
    loading: false,
  });
  return (
    <section className={styles.slideInfo}>
      <Title title={["Under Construction"]} showBg={true}>
        <p>
          My website is currently undergoing .
          <br />
          Should be back shortly, thanks you for you patience.
        </p>
      </Title>
      <form onSubmit={onSendEmail}>
        <input
          required
          disabled={loading}
          type="email"
          name="email"
          placeholder="Enter your email"
          className={styles.email}
          value={emailInput || ""}
          onChange={onChangeForm}
        ></input>
        <button
          className={styles.btnEmail}
          onClick={onSendEmail}
          type="submit"
          name="btn"
        >
          {btn}
        </button>
      </form>
    </section>
  );
}
