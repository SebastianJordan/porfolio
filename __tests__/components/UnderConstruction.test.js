import { render } from '@testing-library/react';
import { UnderConstruction } from '../../components/UnderConstruction';
import '@testing-library/jest-dom';

describe('UnderConstruction', () => {
  test('renders a heading', () => {
    const tree = render(<UnderConstruction />);
    expect(tree).toMatchSnapshot();
  });
});
