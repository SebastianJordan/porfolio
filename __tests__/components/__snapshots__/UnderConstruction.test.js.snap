// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnderConstruction renders a heading 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="slideInfo"
      >
        <div
          class="groupPoint"
        />
        <h1
          class="title"
        >
          Under Construction
          <br />
        </h1>
        <div
          class="description"
        >
          <p>
            My website is currently undergoing .
            <br />
            Should be back shortly, thanks you for you patience.
          </p>
        </div>
        <form>
          <input
            class="email"
            name="email"
            placeholder="Enter your email"
            required=""
            type="email"
            value=""
          />
          <button
            class="btnEmail"
            name="btn"
            type="submit"
          >
            Notify me
          </button>
        </form>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="slideInfo"
    >
      <div
        class="groupPoint"
      />
      <h1
        class="title"
      >
        Under Construction
        <br />
      </h1>
      <div
        class="description"
      >
        <p>
          My website is currently undergoing .
          <br />
          Should be back shortly, thanks you for you patience.
        </p>
      </div>
      <form>
        <input
          class="email"
          name="email"
          placeholder="Enter your email"
          required=""
          type="email"
          value=""
        />
        <button
          class="btnEmail"
          name="btn"
          type="submit"
        >
          Notify me
        </button>
      </form>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
